torchmdnet/datasets/md17.py

```diff
@@ -1,105 +1,151 @@
-        malonaldehyde="md17_malonaldehyde.npz",
-        naphthalene="md17_naphthalene.npz",
-        salicylic_acid="md17_salicylic.npz",
-        toluene="md17_toluene.npz",
-        uracil="md17_uracil.npz",
-    )
-
-    available_molecules = list(molecule_files.keys())
-
-    def __init__(self, root, transform=None, pre_transform=None, molecules=None):
-        assert molecules is not None, (
-            "Please provide the desired comma separated molecule(s) through"
-            f"'molecules'. Available molecules are {', '.join(MD17.available_molecules)} "
-            "or 'all' to train on the combined dataset."
-        )
-
-        if molecules == "all":
-            molecules = ",".join(MD17.available_molecules)
-        self.molecules = molecules.split(",")
-
-        for mol in self.molecules:
-            if mol not in MD17.available_molecules:
-                raise RuntimeError(f"Molecule '{mol}' does not exist in MD17")
-
-        if len(self.molecules) > 1:
-            rank_zero_warn(
-                "MD17 molecules have different reference energies, "
-                "which is not accounted for during training."
-            )
-
-        super(MD17, self).__init__(root, transform, pre_transform)
-
-        self.offsets = [0]
-        self.data_all, self.slices_all = [], []
-        for path in self.processed_paths:
-            data, slices = torch.load(path)
-            self.data_all.append(data)
-            self.slices_all.append(slices)
-            self.offsets.append(
-                len(slices[list(slices.keys())[0]]) - 1 + self.offsets[-1]
-            )
-
-    def len(self):
-        return sum(
-            len(slices[list(slices.keys())[0]]) - 1 for slices in self.slices_all
-        )
-
-    def get(self, idx):
-        data_idx = 0
-        while data_idx < len(self.data_all) - 1 and idx >= self.offsets[data_idx + 1]:
-            data_idx += 1
-        self.data = self.data_all[data_idx]
-        self.slices = self.slices_all[data_idx]
-        return super(MD17, self).get(idx - self.offsets[data_idx])
 
-    @property
-    def raw_file_names(self):
-        return [MD17.molecule_files[mol] for mol in self.molecules]
+    gdml_url = 'http://quantum-machine.org/gdml/data/npz'
+    revised_url = ('https://archive.materialscloud.org/record/'
+                   'file?filename=rmd17.tar.bz2&record_id=466')
+
+    file_names = {
+        'benzene': 'md17_benzene2017.npz',
+        'uracil': 'md17_uracil.npz',
+        'naphtalene': 'md17_naphthalene.npz',
+        'aspirin': 'md17_aspirin.npz',
+        'salicylic acid': 'md17_salicylic.npz',
+        'malonaldehyde': 'md17_malonaldehyde.npz',
+        'ethanol': 'md17_ethanol.npz',
+        'toluene': 'md17_toluene.npz',
+        'paracetamol': 'paracetamol_dft.npz',
+        'azobenzene': 'azobenzene_dft.npz',
+        'revised benzene': 'rmd17_benzene.npz',
+        'revised uracil': 'rmd17_uracil.npz',
+        'revised naphthalene': 'rmd17_naphthalene.npz',
```

Comment:
If I understood correctly, it's going to create a dir (in `self.root`) for each dataset using the key, so maybe use the underscore for the keys: 'revised_benzene', 'revised_uracil', etc.